rfc/0012-policy-report.md

```diff
@@ -95,12 +96,429 @@ results from Kubernetes policy engines (like Kubewarden or Kyverno) but can
 also be used to record audit results from other kind of tools (like Kubernetes
 security scanning tools as Falco or Kubebench).
 
+Kubewarden tailors its audit reports to the scope of the resources it
+evaluates. Depending on the resource's scope, it generates two types of
+reports: `PolicyReport` or  `ClusterPolicyReport`.
+
+Both the `ClusterPolicyReport` and `PolicyReport` serve similar purposes and
+are similar in structure. The primary distinction between them lies in their
+applicability: one is suited for namespaced resources, the other
+for cluster-wide resources. Despite this difference, the
+fundamental aspects and the details they convey remain consistent across both
+report types. So, the following descriptions and interpretations for
+the `PolicyReport` are equally applicable to the `ClusterPolicyReport`.
+
+## Report
+
+When auditing a resource, certain Kubewarden policies may be relevant to the
+events occurring within it. Additionally, the same Kubernetes resource might
+undergo evaluation by multiple policies, resulting in several policy
+evaluations being conducted
+
+The `PolicyReport` provides these information:
+
+- The number of policy evaluations conducted during the audit run.
+- The count of policy evaluations resulting in resource acceptance, rejection,
+  or failure due to errors. Policy failures occur when errors arise during
+  evaluation, such as runtime panics.
+- Comprehensive details regarding all evaluations performed, including the
+  policy utilized, the evaluated resource, the final outcome (acceptance,
+  rejection, or failure), and any accompanying messages returned by the policy.
+
+
+The `PolicyReport` is kept updated during the lifetime of the audited resource,
+meaning:
+
+- New Policies might become "interested" about the resource. New evaluations
+  results will be added to the `PolicyReport`
+- Existing Kubernetes objects and policies can change over time: some
+  evaluations could change their outcome (pass -> fail and vice versa). The
+  evaluation results must reflect that
+- Policies might be deleted or might be no longer "interested" in the resource:
+  the report must be updated to not contain any reference to these policies
+  and their evaluation results
+- Kubernetes resources might be deleted: the report should be deleted together
+  with if
+
+**Note:** `PolicyReport` resources contain security/compliance information. Regular
+users of the cluster should only have READ access to them. They should not be
+able to create, delete or change their contents. This can be done with Kubernetes
+RBAC rules.
```

Comment:
I think we should point out that, due to security reasons, a cluster administrator should play safe and give no access to these reports to regular users of the cluster. If they want to relax that, the could give READ access only to the reports that are defined inside of the Namespaces that can be used by the regular users.

What I mean is: if Bob can create resources under the `foo` Namespace, then Bob might get READ access to the reports inside of the `foo` namespace.